src/components/location-detail/location-services.tsx

```diff
@@ -97,7 +111,14 @@ export default function LocationServices({
             serviceInfo={servicesWrapper}
             name={CATEGORY_DESCRIPTION_MAP[serviceCategory]}
             icon={CATEGORY_ICON_SRC_MAP[serviceCategory]}
-            startExpanded={serviceCategory === previousCategory}
+            startExpanded={
+              serviceCategory === previousSubcategory ||
+              (serviceCategory === previousCategory && !previousSubcategory) ||
+              (previousCategory &&
+                previousCategory === serviceCategory &&
```

Comment:
previousCategory -> previousCategoryAndSubcategoryAsCategory

```suggestion
              (previousCategoryAndSubcategoryAsCategory &&
                previousCategoryAndSubcategoryAsCategory === serviceCategory &&
```